src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -161,10 +161,15 @@ public SolutionState WithNewWorkspace(Workspace workspace, int workspaceVersion)
         // [Conditional("DEBUG")]
         private void CheckInvariants()
         {
-            Contract.ThrowIfTrue(ProjectIds.Count != _projectIdToProjectStateMap.Count);
+            Contract.ThrowIfFalse(_projectIdToProjectStateMap.Count == ProjectIds.Count);
+            Contract.ThrowIfFalse(_projectIdToProjectStateMap.Count == _dependencyGraph.ProjectIds.Count);
```

Comment:
This was the only use for ProjectDependencyGraph.ProjectIds?

Reply:
No, it's public API.